Assets/Shopify/UIToolkit/Themes/CollectionTheme.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Shopify.Unity;
+
+namespace Shopify.UIToolkit {
+    /// <summary>
+    /// A base class for a theme that sells products from a single collection
+    /// </summary>
+    public abstract class CollectionTheme : ThemeBase {
+        /// <summary>
+        /// Called when the implementor should show the UI for selecting products from a list.
+        /// </summary>
+        /// <param name="products">The products that the UI should list</param>
+        public abstract void ShouldShowProductList(Product[] products);
+
+        /// <summary>
+        /// Called when the implementor should show the UI for a particular product and its associated variants.
+        /// </summary>
+        /// <param name="product">The product to show details for</param>
+        /// <param name="variants">The variants of the product</param>
+        public abstract void OnShouldShowProductDetails(Product product, ProductVariant[] variants);
+
+        /// <summary>
+        /// Called when the implementor should show the cart UI with the provided line items.
+        /// </summary>
+        /// <param name="lineItems">The line items to display in the cart</param>
+        public abstract void OnShouldShowCart(CheckoutLineItem[] lineItems);
```

Comment:
Worth noting that im using CheckoutLineItems here but this is subject to change... depends on how we implement the cart functionality... but ill leave this in for now as is